lib/puppet/application.rb

```diff
@@ -417,8 +417,8 @@ def handle_logdest_arg(arg)
       Puppet[:logdest] = arg
       Puppet::Util::Log.newdestination(arg)
       options[:setdest] = true
-    rescue => detail
-      Puppet.log_exception(detail)
+    rescue StandardError => detail
```

Comment:
nit we usually drop `StandardError` as that's the behavior of `rescue => detail`.